cordapp/README.md

```diff
@@ -95,7 +107,337 @@ GqO4M1KlfimphQwCICiq00hDanT5W8bTLqE7GIGuplf/O8AABlpWrUg6uiUB
+
+The identity is set up by the `EnclaveFlowInitiator` class during authentication and sent to the enclave which
+verifies it. Please be aware that the first byte of the identity message indicates whether a party wants to remain
+anonymous or not. If a party decides to remain anonymous, the identity message is padded with zeros to prevent
+anyone in the middle from using statistical analysis to guess whether a party is anonymous or not.
+The remaining bytes represent the party's identity If the party decides to authenticate
+itself.
+
+```kotlin
+@Suspendable
+private fun buildMailerIdentity(): SenderIdentityImpl {
+    val sharedSecret = encryptionKey.publicKey.encoded
+    val signerPublicKey = flow.ourIdentity.owningKey
+    val signature = flow.serviceHub.keyManagementService.sign(sharedSecret, signerPublicKey).withoutKey()
+    val signerCertPath = flow.ourIdentityAndCert.certPath
+    return SenderIdentityImpl(signerCertPath, signature.bytes)
+}
+
+@Suspendable
+@Throws(FlowException::class)
+fun sendIdentityToEnclave(isAnonymous: Boolean) {
+
+    val serializedIdentity = getSerializedIdentity(isAnonymous)
+    sendToEnclave(serializedIdentity)
+
+    val mail: EnclaveMail = session.receive(ByteArray::class.java).unwrap { mail: ByteArray ->
+        try {
+            postOffice.decryptMail(mail)
+        } catch (e: IOException) {
+            throw FlowException("Unable to decrypt mail from Enclave", e)
+        }
+    }
+
+    if (!mail.topic.contentEquals("$flowTopic-ack"))
+        throw FlowException("The enclave could not validate the identity sent")
+}
+```
+
+When the enclave successfully validates the identity, it stores it in a key-based cache. Subsequent messages
+from the same sender in the same session are paired with the cached identity which is then available from
+within the `receiveMail` method through the extra parameter called `identity`. This identity can be used to uniquely identify a sender if the user is not anonymous.
+
+```java
+@Override
+protected void receiveMail(long id, EnclaveMail mail, String routingHint, SenderIdentity identity) {
+    String reversedString = reverse(new String(mail.getBodyAsBytes()));
+
+    String responseString;
+    if (identity == null) {
+        responseString = String.format("Reversed string: %s; Sender name: <Anonymous>", reversedString);
+    } else {
+        responseString = String.format("Reversed string: %s; Sender name: %s", reversedString, identity.getName());
+    }
+
+    // Get the PostOffice instance for responding back to this mail. Our response will use the same topic.
+    final EnclavePostOffice postOffice = postOffice(mail);
+    // Create the encrypted response and send it back to the sender.
+    final byte[] reply = postOffice.encryptMail(responseString.getBytes(StandardCharsets.UTF_8));
+    postMail(reply, routingHint);
+}
+```
+
+The method `receiveMail` contains the enclave logic. You can code your business requirements in the `receiveMail method`.
```

Comment:
I would change the wording of the second sentence a bit to join the sentences together. Something like:
_You can code (or write?) your business requirements in this method too._

Reply:
Changed as:
_You can code your business requirements in this method._